src/App.jsx

```diff
@@ -1,90 +1,15 @@
-import React, { useCallback, useState } from "react";
+import React, { useState } from "react";
 import ClassCheckBox from "./components/ClassCheckbox";
+import classes from "./constants";
 import "./App.css";
 
-const classes = [
-	{
-		id: "class-9",
-		name: "Class 9",
-		checked: false,
-		sections: [
-			{
-				id: "sec-a",
-				name: "Section A",
-				checked: false,
-				students: [
-					{
-						id: "s-1",
-						name: "Student 1",
-						checked: false,
-					},
-					{
-						id: "s-2",
-						name: "Student 2",
-						checked: false,
-					},
-					{
-						id: "s-3",
-						name: "Student 3",
-						checked: false,
-					},
-				],
-			},
-			{
-				id: "sec-b",
-				name: "Section B",
-				checked: false,
-				students: [
-					{
-						id: "s-4",
-						name: "Student 4",
-						checked: false,
-					},
-					{
-						id: "s-5",
-						name: "Student 5",
-						checked: false,
-					},
-					{
-						id: "s-6",
-						name: "Student 6",
-						checked: false,
-					},
-				],
-			},
-			{
-				id: "sec-c",
-				name: "Section C",
-				checked: false,
-				students: [
-					{
-						id: "s-7",
-						name: "Student 7",
-						checked: false,
-					},
-					{
-						id: "s-8",
-						name: "Student 8",
-						checked: false,
-					},
-					{
-						id: "s-9",
-						name: "Student 9",
-						checked: false,
-					},
-				],
-			},
-		],
-	},
-];
 
 export default function App() {
 	const [data, setData] = useState(classes);
 
-	const handleClassCheck = (classId) => {
+	const handleClassCheck = (classIndex) => {
 		let dataClone = [...data];
-		let targetClass = dataClone.find((cls) => cls.id === classId);
-		let classIndex = dataClone.findIndex((cls) => cls.id === classId);
+		let targetClass = dataClone[classIndex];
 
 		targetClass.checked = !targetClass.checked;
```

Comment:
yes but u are making a variable and when the code gets compiled. This variable will also take memory space right? and since u have given the index name as class index . i guess its readable and one can easily understand.

Reply:
ok will make the changes